### ISSUE TITLE
Remove invalid characters from auto-created usernames

### ISSUE THREAD

**Issue by LeoMcA** [mozilla/sumo-project#739]
https://sentry.prod.mozaws.net/operations/sumo-prod/issues/10433280/

This poor user has a `/` in the local part of their email-address, like "foo/bar@example.com", which is a totally valid email address: https://en.wikipedia.org/wiki/Email_address#Local-part

Unfortunately our routing code can't handle that being part of a username, and indeed it shouldn't have to because the username field shouldn't accept it as a valid username: https://docs.djangoproject.com/en/2.2/ref/contrib/auth/#django.contrib.auth.models.User.username

> Usernames may contain alphanumeric, _, @, +, . and - characters.

django-mozilla-oidc calls [`create_user`](https://docs.djangoproject.com/en/2.2/ref/contrib/auth/#django.contrib.auth.models.UserManager.create_user) which doesn't seem to validate the username, which is how this happened.

We should be able to filter out invalid characters in the [`suggest_username`](https://github.com/mozilla/kitsune/blob/30cb647d8013f14ae2b6fb56435bd26b1e88f35d/kitsune/users/utils.py#L37) function which is called when creating users, and we should be able to write a migration to remove invalid chars from all users' usernames.

**Comment by LeoMcA** [comment]
Another example of this: https://sentry.prod.mozaws.net/operations/sumo-prod/issues/10433279/

**Comment by LeoMcA** [comment]
https://github.com/mozilla/kitsune/pull/4695

**Comment by mirunacurtean** [comment]
Verified - as well as can be. 
I've tried and managed to block my account sign up several times with @mozilla.com and @mailinator.com addresses that don't accept many or any special characters in the local part of the email address, so it was not possible for me to produce an auto-created username that used such characters in the first place (Outside of the name+1@mozilla.com examples which were functional and created a username that had no special characters even before)
![image](https://user-images.githubusercontent.com/53306802/109186830-fd66b380-7799-11eb-82e9-3883e5715854.png)

On the profile page instead I did a recheck the username field with all the accepted characters and the invalid ones as well, adding and removing them manually and the results were the expected ones. 
Valid- accepted, the profile is updated. 

![valid username characters](https://user-images.githubusercontent.com/53306802/109188140-4703ce00-779b-11eb-9ed9-1edf88dd6797.PNG)

Invalid ,/?\|';:{}[]""=()*&^%$#! - error message received and profile is not updated. 
![Cannot add back slash](https://user-images.githubusercontent.com/53306802/109188166-4f5c0900-779b-11eb-9ef2-64a654bc8680.PNG)
